### DISCUSSION TITLE
Supporting HTML table structure

### DISCUSSION THREAD

**Discussion by TelosPO** [usnistgov/OSCAL#1589] [Q&A] [answered]
Can the validator read a HTML table structure when expecting Multi-line Markup and string?

**Comment by nikitawootten-nist** [comment, marked as answer]
It can! The Markup Multi-line datatype is described in detail on [the metaschema syntax page](https://pages.nist.gov/metaschema/specification/syntax/#markup-data-types), which includes a [section on tables](https://pages.nist.gov/metaschema/specification/syntax/#tables).

In short,
* The JSON version of Markup Multi-Line is nominally validated against a subset of the [CommonMark spec](https://spec.commonmark.org/), with specific extensions for things like tables which are not a part of the core specification.
* The XML version of Markup Multi-Line is represented as a subset of HTML that maps cleanly to CommonMark.
* Content converters such as `liboscal-java` are responsible for mapping between the JSON's CommonMark representation and the XML's HTML subset.
* Commonmark allows for insertion of the same subset of HTML elements (like `<table>`), meaning that **you can insert an HTML table into either a JSON or XML document**. When using inline HTML in Markdown, you will lose the conversion guarantees between XML<->JSON/YAML.
